### PR TITLE
Autoupdate

### PR DESCRIPTION
**Pull request by romangraef**

### CODE REVIEW

**Comment by romangraef** [comment]
@IRONM00N can you merge this, infer is just picking up on all the other bugs in the same file that i havent caused (and probably arent critical either)

File: src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java:52]: this seems like a bad idea

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/updater/UpdateLoader.java
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/updater/UpdateLoader.java:110]: 🙄

**Review by IRONM00N — APPROVED** [review]

**Comment by IRONM00N** [comment]
> @IRONM00N can you merge this, infer is just picking up on all the other bugs in the same file that i havent caused (and probably arent critical either)

I would prefer if at least one other person approves first. Especially considering this is an auto updater.

File: src/main/java/io/github/moulberry/notenoughupdates/NEUOverlay.java
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/NEUOverlay.java:1482]: Also, what do synchronized methods do?

File: src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java
Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java:52]: what?

**Comment by romangraef** [comment]
yeah i want other people to approve, just once there are approvals, can you merge this despite infer failing cause im not fixing all those alleged race conditions in the search bar

**Comment by IRONM00N** [comment]
> yeah i want other people to approve, just once there are approvals, can you merge this despite infer failing cause im not fixing all those alleged race conditions in the search bar

yea I can merge it then

**Review by NopoTheGamer — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Review by DoKM — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Review by DoKM — APPROVED** [review]

**Pull request merged**
